src/parcsr_ls/par_ilu_setup_device.c

```diff
@@ -373,4 +416,190 @@ hypre_ILUSetupILUDevice(HYPRE_Int               ilu_type,
    return hypre_error_flag;
 }
 
+/*--------------------------------------------------------------------------
+ * hypre_ILUSetupIterativeILU0Device
+ *
+ * This function computes an ILU0 iteratively with rocSPARSE.
+ *
+ * Input arguments:
+ *   A - input matrix
+ *   type - algorithm for computing iterative ILU0
+ *   option - internal flags used by rocSPARSE
+ *   max_iter - max. number of iterations
+ *   tolerance - stopping criteria in iterative algorithm
+ *
+ * Output arguments:
+ *   num_iter_ptr - number of iterations
+ *   history_ptr - list of corrections and residual values for each iteration
+ *                 (computed only when residuals_ptr points to NULL at entrance)
```

Comment:
Do you mean if the history_ptr points to NULL here (instead of residuals_ptr? Also, the logic seems a bit confusing for me. We want to populate history information if the pointer is NULL. What if it is not? Would we not want updated info?

Reply:
Done in [951faeb](https://github.com/hypre-space/hypre/pull/1011/commits/951faeb683eef098643fae4b2501db948c84f021)